cmd/kubernetes_app_remove.go

```diff
@@ -0,0 +1,94 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/civo/civogo"
+	"github.com/civo/cli/config"
+	"github.com/civo/cli/utility"
+	"github.com/spf13/cobra"
+)
+
+var kubernetesAppRemoveCmd = &cobra.Command{
+	Use:     "remove",
+	Example: "civo kubernetes application remove NAME --cluster CLUSTER_NAME",
+	Aliases: []string{"uninstall"},
+	Args:    cobra.MinimumNArgs(1),
+	Short:   "Remove the marketplace application to a Kubernetes cluster",
+	Run: func(cmd *cobra.Command, args []string) {
+		utility.EnsureCurrentRegion()
+
+		client, err := config.CivoAPIClient()
+		if regionSet != "" {
+			client.Region = regionSet
+		}
+		if err != nil {
+			utility.Error("Creating the connection to Civo's API failed with %s", err)
+			os.Exit(1)
+		}
+		kube, err := client.FindKubernetesCluster(kubernetesClusterApp)
+		if err != nil {
+			utility.Error("%s", err)
+			os.Exit(1)
+		}
+		if !kube.Ready {
+			utility.Error("The cluster isn't ready yet, so the KUBECONFIG isn't available.")
+			os.Exit(1)
+		}
+
+		allsplit := strings.Split(args[0], ",")
+		tmpFile, err := ioutil.TempFile(os.TempDir(), "kubeconfig-")
+		if err != nil {
+			utility.Error("Cannot create temporary file", err)
+		}
+		if _, err = tmpFile.Write([]byte(kube.KubeConfig)); err != nil {
+			utility.Error("Failed to write to temporary file", err)
+		}
+		defer os.Remove(tmpFile.Name())
+		for _, split := range allsplit {
+			appName := split
+			// TODO: Ideally this would come from the Civo API, but the Civo API doesn't currently return uninstall.sh
+			// https://www.civo.com/api/kubernetes#listing-applications
+			filepath := fmt.Sprintf("bash <(curl -s https://raw.githubusercontent.com/civo/kubernetes-marketplace/master/%s/uninstall.sh)", appName)
+			cmd_config := exec.Command("/bin/bash", "-c", filepath)
+			var b bytes.Buffer
+			cmd_config.Stdout = &b
+			cmd_config.Stderr = &b
+			cmd_config.Env = os.Environ()
+			cmd_config.Env = append(cmd_config.Env, "KUBECONFIG="+tmpFile.Name())
+
+			if err := cmd_config.Run(); err != nil {
+				if exitError, ok := err.(*exec.ExitError); ok {
+					fmt.Printf("Failed to uninstall application %s (exited with code %d)\n", appName, exitError.ExitCode())
+					cmd := exec.Command("curl", "-s", fmt.Sprintf("https://raw.githubusercontent.com/civo/kubernetes-marketplace/master/%s/uninstall.sh", appName))
+					output, _ := cmd.CombinedOutput()
+					fmt.Println("--------------- Uninstall script ---------------")
+					fmt.Println(string(output))
+					fmt.Println("--------------- Uninstall output ---------------")
+					fmt.Println(b.String())
+					os.Exit(1)
+				} else {
+					fmt.Printf("Failed to uninstall application %s because of %s\n", appName, err.Error())
```

Comment:
Can we use `utility.Error()` here? Just to keep our logging as consistent as possible

Reply:
> @saiyam1814 when testing this, I get the following error -
> 
> ```
> haardik@Haardiks-MacBook-Pro dest % ./civo_mac kubernetes show new-cluster                              
>           ID : 6154086b-b504-4eec-8860-6e40151fd6f9
>         Name : new-cluster
>       Region : LON1
>        Nodes : 1
>         Size : g4s.kube.xsmall
>       Status : ACTIVE
>      Version : 1.22.2-k3s1
> API Endpoint : https://74.220.18.31:6443
>  External IP : 74.220.18.31
> DNS A record : 6154086b-b504-4eec-8860-6e40151fd6f9.k8s.civo.com
> 
> Pool (c8f2ad):
> +--------------------------------------------+--------------+--------+-----------------+-----------+----------+---------------+
> | Name                                       | IP           | Status | Size            | Cpu Cores | RAM (MB) | SSD disk (GB) |
> +--------------------------------------------+--------------+--------+-----------------+-----------+----------+---------------+
> | k3s-new-cluster-76d4-0b0c52-node-pool-4ac0 | 74.220.18.31 | ACTIVE | g4s.kube.xsmall |         1 |     1024 |            30 |
> +--------------------------------------------+--------------+--------+-----------------+-----------+----------+---------------+
> 
> Labels:
> kubernetes.civo.com/node-pool=c8f2ad4f-c4fe-49d9-bd55-f62426a6e90c
> kubernetes.civo.com/node-size=g4s.kube.xsmall
> 
> 
> Applications:
> +-------+---------+-----------+--------------+
> | Name  | Version | Installed | Category     |
> +-------+---------+-----------+--------------+
> | kafka | 2.7.0   | true      | architecture |
> +-------+---------+-----------+--------------+
> 
> haardik@Haardiks-MacBook-Pro dest % ./civo_mac kubernetes apps remove --help
> Remove the marketplace application to a Kubernetes cluster
> 
> Usage:
>   civo kubernetes applications remove [flags]
> 
> Aliases:
>   remove, uninstall
> 
> Examples:
> civo kubernetes application remove NAME --cluster CLUSTER_NAME
> 
> Flags:
>   -c, --cluster string   the name of the cluster to remove the app.
>   -h, --help             help for remove
> 
> Global Flags:
>       --config string   config file (default is $HOME/.civo.json)
>   -f, --fields string   output fields for custom format output (use -h to determine fields)
>   -o, --output string   output format (json/human/custom) (default "human")
>       --pretty          Print pretty the json output
>       --region string   Choose the region to connect to, if you use this option it will use it over the default region
>   -y, --yes             Automatic yes to prompts; assume "yes" as answer to all prompts and run non-interactively
> 
> 
> haardik@Haardiks-MacBook-Pro dest % ./civo_mac kubernetes applications remove kafka --cluster new-cluster
> Failed to uninstall application kafka (exited with code 1)
> --------------- Uninstall script ---------------
> #!/bin/bash
> 
> # Delete the app
> helm uninstall kafka -n kafka
> 
> # Delete the namespace
> kubectl delete namespace kafka
> 
> --------------- Uninstall output ---------------
> /dev/fd/63: line 4: helm: command not found
> Error from server (NotFound): namespaces "kafka" not found
> ```

This might occur as helm uninstall will delete the ns already? that would anyways be the uninstall.sh fix rather than here IMO.